BraveShared/CertificatePinning.swift

```diff
@@ -86,14 +89,27 @@ public class PinningCertificateEvaluator: NSObject, URLSessionDelegate {
             throw error(reason: "Empty Certificates")
         }
         
-        // Self signed anchoring
-        if options.contains(.allowSelfSigned) {
+        // Certificate anchoring
+        if options.contains(.anchorSpecificTrustsOnly) || options.contains(.anchorSpecificAndSystemTrusts) {
+            // Add the certificates to the trust
             guard SecTrustSetAnchorCertificates(trust, certificates as CFArray) == errSecSuccess else {
-                throw error(reason: "Self Signed Certificate Anchor Failed")
+                throw error(reason: "Certificate Anchor Failed")
             }
             
-            guard SecTrustSetAnchorCertificatesOnly(trust, true) == errSecSuccess else {
-                throw error(reason: "Self Signed Certificate Anchor Only Failed")
+            if options.contains(.anchorSpecificTrustsOnly) {
+                // Trust only the passed in certificates (true)
+                //
+                // This is the default behaviour, however we do it explicitly to throw an exception
+                // immediately upon failure
+                // The default behaviour will silently ignore the exception until validation
+                guard SecTrustSetAnchorCertificatesOnly(trust, true) == errSecSuccess else {
+                    throw error(reason: "Self-Signed Certificate Anchor Only Failed")
```

Comment:
I'm confused about the "self-signed" bit here. Which certs are self-signed? The pinned root CAs?

Reply:
Only the one in the unit test is self-signed. 

Self-Signed Certificate:
![image](https://user-images.githubusercontent.com/1530031/135866493-153fb842-a29f-45cd-87af-a2e30afdecd4.png)